packages/tools/tests/test/visualization/config.json

```diff
@@ -1488,6 +1488,18 @@
             "referenceImage": "parseGuiJsonWithUnicode.png",
             "renderCount": 50
         },
+        {
+            "title": "InputManager callback test",
+            "playgroundId": "#30ZGX9#1",
+            "referenceImage": "inputManagerCallbackTest.png",
+            "renderCount": 10

```

Comment:
It would also be great at the DeviceInputSystem level to simulate the event you are doing at the canvas level in the Playgrounds.

This way we could have simply unit tests it instead of slower visualization tests.

We could probably simulate being a native input system to trigger it all by code ;-)

Reply:
_Is it possible to make the PG not be ready until the picking it complete instead of using renderCount?_
There aren't any options in the testing JSON to facilitate this, AFAIK.  Would it be possible to use something like `delayCreateScene` to do this?

_It would also be great at the DeviceInputSystem level to simulate the event you are doing at the canvas level in the Playgrounds._
I know that in our Unit Tests, we have a version of the DeviceInputSystem that can take a changed input, create an event object, and dispatch it.  That can be used, in a sense to simulate dispatching an event without actually having the web context.  As of right now, it only accepts a single change (eg. `PointerInput.LeftClick`) but could be modified to be closer to what an event would send (eg. both a button press and x/y changes).

Reply:
> There aren't any options in the testing JSON to facilitate this, AFAIK. Would it be possible to use something like `delayCreateScene` to do this?

The validation tests wait for the scene to be ready. You can leverage that mechanism or make the createScene an async function and return a promise.